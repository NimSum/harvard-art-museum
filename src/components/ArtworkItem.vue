<template>
  <article>
    <h2>Work</h2>
  </article>
</template>

<script>
export default {
  name: 'ArtworkItem',
  props: ['artwork'],
  data() {
    return {
      images: []
    }
  },
  created() {
    const { people, colors, dated, labeltext, primaryimageurl, url } = this.$props.artwork;
    console.log(dated)
  }
}
</script>

<style>

</style>
