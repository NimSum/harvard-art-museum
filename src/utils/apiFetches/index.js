import axios from 'axios';

const getAnything = async (url) => {
  console.log('fetching')
  const response = await axios.get(url);
  if (response.statusText !== 'OK') {
    throw new Error('Failed to fetch data');
  }
  return response.data;
}

export default getAnything;