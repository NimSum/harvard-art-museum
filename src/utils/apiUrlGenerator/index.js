const apiUrls = {
  galleriesByFloor: (floor) => `https://api.harvardartmuseums.org/gallery?floor=${floor}&hasimage=1&apikey=${process.env.VUE_APP_API_KEY}&size=15&fields=galleryid,name,theme,objectcount,labeltext`,
  getGalleryObjects: (galleryId) => `https://api.harvardartmuseums.org/object?apikey=${process.env.VUE_APP_API_KEY}&sort=random&gallery=${galleryId}`,
  randomImages: (randomStart) => `https://api.harvardartmuseums.org/image?apikey=${process.env.VUE_APP_API_KEY}&size=21&sort=random&page=${randomStart}` 
}

export default apiUrls;