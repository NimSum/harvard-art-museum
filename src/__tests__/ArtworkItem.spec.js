import { mount } from '@vue/test-utils';
import ArtworkItem from '../components/ArtworkItem.vue';
import { gallery1220 } from '../utils/mockData';

describe('ArtworkItem', () => {
  const mockArtwork = gallery1220.records[1];

  test('renders correctly', () => {
    const wrapper = mount(ArtworkItem, {
        propsData: {
          artwork: mockArtwork
        }
      }
    );
    expect(wrapper.element).toMatchSnapshot();
  })
});