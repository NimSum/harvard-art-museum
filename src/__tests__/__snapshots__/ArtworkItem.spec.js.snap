// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkItem renders correctly 1`] = `
<article
  class="ArtworkItem"
>
  <div
    class="image-container"
  >
    <img
      alt=""
      src="https://nrs.harvard.edu/urn-3:HUAM:DDC251912_dynmc"
    />
     
    <h4>
      Gabrielle in a Red Dress
    </h4>
     
    <button
      type="button"
    >
      More Details
    </button>
  </div>
   
  <div
    class="details-container"
    style="display: none;"
  >
    <table>
      <tr>
        <td>
          Dated:
        </td>
         
        <td>
          1908
        </td>
      </tr>
       
      <tr>
        <td>
          Medium: 
        </td>
         
        <td>
          Oil on canvas
        </td>
      </tr>
       
      <tr>
        <td>
          Culture: 
        </td>
         
        <td>
          French
        </td>
      </tr>
       
      <tr>
        <td>
          Artist/Artists: 
        </td>
         
        <td>
          Pierre-Auguste Renoir
        </td>
      </tr>
    </table>
     
    <a
      href="https://www.harvardartmuseums.org/collections/object/229048"
      rel="noopener noreferrer"
      target="_blank"
    >
      
      Harvard Museum Details
    
    </a>
  </div>
   
  <div
    class="details-container"
  >
    <p>
      Renoir painted Gabrielle Renard (1879–1959) more than two hundred times. In several of the portraits, she wears the same informal, square-necked gown seen here. By 1908, she had been employed in Renoir’s household for fourteen years, as a nanny, housekeeper, model, and companion to the aging artist.
This painting’s gold and brownish-red palette, loose brushwork, and stylized, heavy forms typify Renoir’s late work; he adopted the style in response to the ancient murals of Pompeii and Herculaneum, which he had seen in Naples. His direct engagement with the classical past was greatly admired, and was seen as a modern project by many young avant-garde artists, including Picasso and Matisse. The latter visited Renoir frequently in his final years, and praised him in the highest terms: “I have always felt that recorded time holds no nobler story, no more heroic, no more magnificent achievement than that of Renoir.”
    </p>
  </div>
</article>
`;
